### PR TITLE
Remove $viewBasePath from ViewRenderer, add documentation

### PR DESCRIPTION
**Pull request by devanych**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

File: src/ViewRenderer.php [hunk 1]
```diff
@@ -37,51 +43,61 @@ final class ViewRenderer implements ViewContextInterface
...
         $this->view = $view;
-
-        $this->viewBasePath = $viewBasePath;
+        $this->viewPath = rtrim($viewPath, '/');
         $this->layout = $layout;
         $this->injections = $injections;
     }
 
+    /**
+     * Returns a path to the directory of views that is a prefix to the relative view name.
```
Comment (samdark) [inline src/ViewRenderer.php:81]:
```suggestion
     * Returns a path to a base directory of view templates that is prefixed to the relative view name.
```

File: src/ViewRenderer.php [hunk 2]
```diff
@@ -37,51 +43,61 @@ final class ViewRenderer implements ViewContextInterface
...
+     *
+     * If a controller name has been set {@see withController(), withControllerName()}, it will be appended to the path.
```
Comment (samdark) [inline src/ViewRenderer.php:83]: Prepended?

File: src/ViewRenderer.php [hunk 3]
```diff
@@ -37,51 +43,61 @@ final class ViewRenderer implements ViewContextInterface
...
+     *
+     * @return string The path to the directory of views that is a prefix to the relative view name.
```
Comment (samdark) [inline src/ViewRenderer.php:85]:
```suggestion
     * @return string View templates base directory.
```

File: src/ViewRenderer.php [hunk 4]
```diff
@@ -37,51 +43,61 @@ final class ViewRenderer implements ViewContextInterface
...
-        if ($this->viewPath !== null) {
-            return $this->aliases->get($this->viewPath);
-        }
-
-        return $this->aliases->get($this->viewBasePath) . ($this->name ? '/' . $this->name : '');
+        return $this->aliases->get($this->viewPath) . ($this->name ? '/' . $this->name : '');
     }
 
+    /**
+     * Renders a view and places it in the response instance.
```
Comment (samdark) [inline src/ViewRenderer.php:93]:
```suggestion
     * Renders a view returns a response instance.
```

File: src/ViewRenderer.php [hunk 5]
```diff
@@ -100,6 +116,15 @@ public function render(string $view, array $parameters = []): ResponseInterface
         return $this->responseFactory->createResponse($contentRenderer);
     }
 
+    /**
+     * Renders a view without applying a layout and places it in the response instance.
```
Comment (samdark) [inline src/ViewRenderer.php:120]:
```suggestion
     * Renders a view without applying a layout and returns a response instance.
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/ViewRenderer.php (hunk 2 above)
Comment (devanych) [reply src/ViewRenderer.php:83]: No, that's right. The controller name is appended to the views directory.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**